e2e/exec.go

```diff
@@ -0,0 +1,172 @@
+		t.Logf("executing command on remote VM at %s of VMSS %s: %q", privateIP, vmssName, sourceCmd)
+
+		execResult, err := execOnVM(ctx, opts.kube, privateIP, podName, sshPrivateKey, sourceCmd)
+		if execResult != nil {
+			checkStdErr(execResult.stderr, t)
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		result[file] = execResult.stdout.String()
+	}
+	return result, nil
+}
+
+func extractClusterParameters(ctx context.Context, t *testing.T, kube *kubeclient) (map[string]string, error) {
+	commandList := map[string]string{
+		"/etc/kubernetes/azure.json":            "cat /etc/kubernetes/azure.json",
+		"/etc/kubernetes/certs/ca.crt":          "cat /etc/kubernetes/certs/ca.crt",
+		"/var/lib/kubelet/bootstrap-kubeconfig": "cat /var/lib/kubelet/bootstrap-kubeconfig",
+	}
+
+	podName, err := getDebugPodName(kube)
+	if err != nil {
+		return nil, err
+	}
+
+	var result = map[string]string{}
+	for file, sourceCmd := range commandList {
+		t.Logf("executing privileged command on pod %s/%s: %q", defaultNamespace, podName, sourceCmd)
+
+		execResult, err := execPrivilegedOnPod(ctx, kube, defaultNamespace, podName, sourceCmd)
+		if execResult != nil {
+			checkStdErr(execResult.stderr, t)
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		result[file] = execResult.stdout.String()
+	}
+
+	return result, nil
+}
+
+func execOnVM(ctx context.Context, kube *kubeclient, vmPrivateIP, jumpboxPodName, sshPrivateKey, command string) (*podExecResult, error) {
+	sshCommand := fmt.Sprintf(sshCommandTemplate, sshPrivateKey, vmPrivateIP)
+	commandToExecute := fmt.Sprintf("%s %s", sshCommand, command)
+
+	execResult, err := execPrivilegedOnPod(ctx, kube, defaultNamespace, jumpboxPodName, commandToExecute)
+	if err != nil {
+		return nil, fmt.Errorf("error executing command on pod: %s", err)
+	}
+
+	return execResult, nil
+}
+
+func execPrivilegedOnPod(ctx context.Context, kube *kubeclient, namespace, podName string, command string) (*podExecResult, error) {
+	privilegedCommand := append(nsenterCommandArray(), command)
+	return execOnPod(ctx, kube, namespace, podName, privilegedCommand)
+}
+
+func execOnPod(ctx context.Context, kube *kubeclient, namespace, podName string, command []string) (*podExecResult, error) {
+	req := kube.typed.CoreV1().RESTClient().Post().Resource("pods").Name(podName).Namespace(namespace).SubResource("exec")
+
+	option := &corev1.PodExecOptions{
+		Command: command,
+		Stdout:  true,
+		Stderr:  true,
+	}
+
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+
+	exec, err := remotecommand.NewSPDYExecutor(kube.rest, "POST", req.URL())
+	if err != nil {
+		return nil, fmt.Errorf("unable to create new SPDY executor for pod exec: %s", err)
+	}
+
+	var (
+		stdout, stderr bytes.Buffer
+		exitCode       string = "0"
+	)
+
+	err = exec.StreamWithContext(ctx, remotecommand.StreamOptions{
+		Stdout: &stdout,
+		Stderr: &stderr,
+	})
+	if err != nil {
+		if strings.Contains(err.Error(), "command terminated with exit code") {
```

Comment:
yeah this seems to be the pattern people follow from a bit of googling, though I couldn't find another reference of someone needing to extract exit code specifically from stderr or the stream error message directly, though this does make sense since it seems like exit codes are independent from stderr entirely

also makes sense from what I've seen the contents of stderr actually are in previous testing